zaza/openstack/charm_tests/mysql/test_prometheus_mysql_exporter.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2022 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""MySQL Prometheus Exporter Testing."""
+
+import json
+
+import zaza.model as zaza_model
+from zaza.openstack.charm_tests.mysql.tests import MySQLBaseTest
+
+
+class PrometheusMySQLExporterTest(MySQLBaseTest):
+    """Functional tests check prometheus exporter."""
+
+    @classmethod
+    def setUpClass(cls, application_name=None):
+        """Run class setup for running mysql tests."""
+        super().setUpClass(application_name="mysql-innodb-cluster")
+        cls.application = "mysql-innodb-cluster"
+        cls.snap_name = "mysqld-exporter"
+        cls.service_name = "snap.mysqld-exporter.mysqld-exporter.service"
+
+    def _exporter_http_check(
+        self,
+        cmd,
+        expected,
+    ):
+        """Exec check cmd on each unit in the application.
+
+        :param cmd: The check command run on unit
+        :type cmd: str
+        :param expected: Expected result code
+        :type expected: str
+        """
+        for unit in zaza_model.get_units(self.application):
+            result = zaza_model.run_on_unit(unit.name, cmd)
+            self.assertEqual(result.get("Code"), expected)
+
+    def _check_service_status_is(
+        self,
+        active=True,
+    ):
+        cmd = "systemctl is-active {}".format(
+            self.service_name
+        )
+        excepted = "active\n"
+        if not active:
+            excepted = "inactive\n"
+        for unit in zaza_model.get_units(self.application):
+            result = zaza_model.run_on_unit(unit.name, cmd)
+            self.assertEqual(result.get("stdout"), excepted)
+
+    def test_01_exporter_http_check(self):
+        """Check exporter endpoint is working."""
+        self._exporter_http_check(
+            cmd="curl http://localhost:9104",
```

Comment:
Is there a specific reason for logging in and running a curl from localhost here? I think there's value in just querying it remotely.

Reply:
I think we can add another check `mysql_up` = 1 here. It can make sure exporter service is alive also permission is correct.